lib/vtls/bearssl.c

```diff
@@ -100,26 +109,38 @@ static CURLcode load_cafile(const char *path, br_x509_trust_anchor **anchors,
   br_x509_trust_anchor *new_anchors;
   size_t new_anchors_len;
   br_x509_pkey *pkey;
-  FILE *fp;
-  unsigned char buf[BUFSIZ], *p;
+  FILE *fp = 0;
+  unsigned char buf[BUFSIZ];
+  const unsigned char *p;
   const char *name;
   size_t n, i, pushed;
 
-  fp = fopen(path, "rb");
-  if(!fp)
-    return CURLE_SSL_CACERT_BADFILE;
+  if(source->type != CAFILE_SOURCE_PATH && source->type != CAFILE_SOURCE_BLOB)
+    return CURLE_BAD_FUNCTION_ARGUMENT;
```

Comment:
This can't happen can it the way the code is written and intended? Shouldn't this then rather be made with DEBUGASSERT() instead?